source/Halibut/Transport/PollingClient.cs

```diff
@@ -46,37 +46,43 @@ public void Start()
             thread.Start();
         }
 
-        private void ExecutePollingLoop(object ignored)
+        public void Dispose()
+        {
+            working = false;
+        }
+
+        void ExecutePollingLoop(object ignored)
         {
-            var retry = new PhasedBackoffRetryTracker();
+            var retry = RetryPolicy.Create();
+            var sleepFor = TimeSpan.Zero;
             while (working)
             {
                 try
                 {
-                    retry.Try();
-                    secureClient.ExecuteTransaction(protocol =>
+                    try
+                    {
+                        retry.Try();
+                        secureClient.ExecuteTransaction(protocol => { protocol.ExchangeAsSubscriber(subscription, handleIncomingRequest); }, cancellationToken);
+                        retry.Success();
+                    }
+                    finally
```

Comment:
Reading through the previous `while` loop, this appears to address the _"don't DDoS ourselves when we're successful"_ scenario, while also providing a consistent way to get the next sleep period when there _is_ an exception 👍

When last we spoke, we were confused as to whether there was something higher up the chain providing backoff/sleep, but it looks like this has just been constantly hammering things?

Reply:
I still don't know :( I'm having a diabolical time with nested retry loops trying to get test instances to fail in the desired manner. I'm hoping that even if this change doesn't address the entirety of the issue, it will get most of the worst of it.